src/main/java/io/vertx/kafka/client/consumer/impl/KafkaReadStreamImpl.java

```diff
@@ -657,4 +657,20 @@ public KafkaReadStream batchHandler(Handler<ConsumerRecords<K, V>> handler) {
     this.pollTimeout = timeout;
     return this;
   }
+
+  @Override
+  public void poll(long timeout, Handler<AsyncResult<ConsumerRecords<K, V>>> handler) {
+    this.worker.submit(() -> {
+      if (!this.closed.get()) {
+        Future fut;
+        try {
+          ConsumerRecords<K, V> records = this.consumer.poll(timeout);
+          this.context.runOnContext(v -> handler.handle(Future.succeededFuture(records)));
+        } catch (WakeupException ignore) {
```

Comment:
shouldn't signal this with an empty record set ?

Reply:
well in general this exception is raised by a `wakeup()` call on the native consumer which is something that we don't expose here. It's needed for waking up the polling thread due to the synchronous nature of the `poll`. In general the exception is ignored because if you do that you are going to close the consumer (but of course there could be different use cases). Returning an empty records could make sense ... or a failed future with the exception.
Maybe we should expose a `wakeup` method as well ?

Reply:
I think in all case, the asynchronous call expects a callback whatever it is, otherwise it's equivalent to a `Thread.stop()` :-)

why would we need a wakeup method as well ?

Reply:
> why would we need a wakeup method as well ?

could it be useful to having the handler called quicker instead of waiting for a long timeout in the poll? I cannot think about a specific use case though.

Reply:
I think that's rather low level and we want avoid exposing this

Reply:
I pushed the change ;)